omas/utilities/omas_mds.py

```diff
@@ -146,22 +146,18 @@ def mdsk(value):
 
             try:
                 out_results = None
-
-                # try connecting and re-try on fail
+                global current_conn
                 for fallback in [0, 1]:
-                    if (self.server, self.treename, self.pulse) not in _mds_connection_cache:
-                        conn = MDSplus.Connection(self.server)
-                        if self.treename is not None:
-                            conn.openTree(self.treename, self.pulse)
-                        _mds_connection_cache[(self.server, self.treename, self.pulse)] = conn
```

Comment:
Couldn't the overall changes of this PR be accomplished by just changing the key for the cache to `self.server` (and moving some of the logic here)?  I don't like the `current_conn` global; what if a person wants to connect to more than one server simultaneously?